### PR TITLE
#2628 Delete Work Package Template Button

### LINKED ISSUE #2628: [Work Package Template] - Add delete work package template button
### Desired Changes

We need a way for an admin to delete a work package template. We need a delete icon next to each template on the admin tools page. A modal should pop up asking if the user wants to delete the work package template. If so, call the delete work package template endpoint to delete it

### Acceptance Criteria
- There is a delete icon next to each work package template
- When it is clicked, a modal pops up asking the user if they want to delete it
- The useDeleteSingleWorkPackageTemplate hook is called to delete the work package template

### Screenshots (as needed)

_No response_

### PR DESCRIPTION
**Pull request by walker-sean**
## Changes

Added delete buttons for work package templates and a modal to confirm that connects to the hook

## Notes
I also fixed some typos I found

## Screenshots

_If you made UI changes you must post a screenshot of the whole page for each change in 1) a normal sized window and 2) the smallest possible window_

_If you did any manual testing (e.g., with Postman), put screenshots of the http request and before and after of the db_

_If none of this applies, you can delete this section_

image

image

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #2628

### CODE REVIEW

File: src/frontend/src/hooks/work-packages.hooks.ts [hunk 1]
```diff
@@ -167,14 +167,14 @@ export const useAllWorkPackageTemplates = () => {
 export const useDeleteWorkPackageTemplate = () => {
   const queryClient = useQueryClient();
   return useMutation<{ message: string }, Error, string>(
-    ['work package template', 'delete'],
+    ['work package templates', 'delete'],
```
Comment (Peyton-McKee) [inline src/frontend/src/hooks/work-packages.hooks.ts:170]: Can u change the create worn package template funtion below these to have the s as well

File: src/frontend/src/pages/AdminToolsPage/ProjectsConfig/WorkPackageTemplateTable.tsx [hunk 2]
```diff
@@ -16,27 +20,52 @@ const WorkPackageTemplateTable = () => {
...
+        submitText="Delete"
+        onSubmit={() => {
+          mutateAsync(templateToDelete!.workPackageTemplateId);
+          setTemplateToDelete(undefined);
+        }}
+      >
+        <Typography>
+          Are you sure you want to delete the work package template <i>{templateToDelete?.templateName}</i>?
+        </Typography>
+        <Typography>This will also delete all templates blocked by this one.</Typography>
```
Comment (Peyton-McKee) [inline src/frontend/src/pages/AdminToolsPage/ProjectsConfig/WorkPackageTemplateTable.tsx:65]: Add a statement saying if you wish to only delete this work package template, remove any reference to this template from any other templates blocked bys. But put that into chatgpt to be more clear

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Shwag

**Review by Peyton-McKee — APPROVED** [review]
This looks good we probably shouldnt merge it until aaryans is good if u wanna check that out first. Also i think he mightve made the hooks in that one too

**Comment by Aaryan1203** [comment]
> This looks good we probably shouldnt merge it until aaryans is good if u wanna check that out first. Also i think he mightve made the hooks in that one too

Wait I think my hooks are merged in already? @Peyton-McKee

**Comment by Peyton-McKee** [comment]
oh right I forgot i merged that, does that mean we now have two sets of hooks? @walker-sean can you see if theres any duplicates in that file if you merge develop?

**Comment by Peyton-McKee** [comment]
oh nevermind then I missed that

**Pull request merged**